src/main/java/at/hannibal2/skyhanni/api/event/EventListeners.kt

```diff
@@ -68,15 +68,31 @@ class EventListeners private constructor(val name: String, private val isGeneric
     class Listener(
         val name: String,
         val invoker: Consumer<Any>,
-        val options: HandleEvent,
+        val options: HandleEventCached,
         val generic: Class<*>?,
     ) {
         val onlyOnIslandTypes: Set<IslandType> = getIslands(options)
 
         companion object {
-            private fun getIslands(options: HandleEvent): Set<IslandType> =
+            private fun getIslands(options: HandleEventCached): Set<IslandType> =
```

Comment:
Since the data is now being stored in its own class instead of a proxy you can move this Set<> logic in there directly

Reply:
yep like that